sig-scalability/slos/api_call_latency.md

```diff
@@ -5,23 +5,28 @@
 | Status | SLI | SLO |
 | --- | --- | --- |
 | __Official__ | Latency<sup>[1](#footnote1)</sup> of mutating<sup>[2](#footnote2)</sup> API calls for single objects for every (resource, verb) pair, measured as 99th percentile over last 5 minutes | In default Kubernetes installation, for every (resource, verb) pair, excluding virtual and aggregated resources and Custom Resource Definitions, 99th percentile per cluster-day <= 1s |
-| __Official__ | Latency<sup>[1](#footnote1)</sup> of non-streaming read-only<sup>[3](#footnote3)</sup> API calls for every (resource, scope<sup>[4](#footnote4)</sup>) pair, measured as 99th percentile over last 5 minutes | In default Kubernetes installation, for every (resource, scope) pair, excluding virtual and aggregated resources and Custom Resource Definitions, 99th percentile per cluster-day (a) <= 1s if `scope=resource` (b) <= 5s if `scope=namespace` (c) <= 30s if `scope=cluster` |
+| __Official__ | Latency<sup>[1](#footnote1)</sup> of non-streaming read-only<sup>[3](#footnote3)</sup> API calls for every (resource, scope<sup>[4](#footnote4)</sup>) pair, measured as 99th percentile over last 5 minutes | In default Kubernetes installation, for every (resource, scope) pair, excluding virtual and aggregated resources and Custom Resource Definitions, 99th percentile per cluster-day: (a) <= 1s if `scope=resource` (b) <= 30s<sup>[5](#footnote5)</sup> otherwise (if `scope=namespace` or `scope=cluster`) |
 
-<a name="footnote1">\[1\]</a>By latency of API call in this doc we mean time
+<a name="footnote1">\[1\]</a> By latency of API call in this doc we mean time
 from the moment when apiserver gets the request to last byte of response sent
 to the user.
 
-<a name="footnote2">\[2\]</a>By mutating API calls we mean POST, PUT, DELETE
+<a name="footnote2">\[2\]</a> By mutating API calls we mean POST, PUT, DELETE
 and PATCH.
 
-<a name="footnote3">\[3\]</a>By non-streaming read-only API calls we mean GET
+<a name="footnote3">\[3\]</a> By non-streaming read-only API calls we mean GET
 requests without `watch=true` option set. (Note that in Kubernetes internally
 it translates to both GET and LIST calls).
 
-<a name="footnote4">\[4\]</a>A scope of a request can be either (a) `resource`
-if the request is about a single object, (b) `namespace` if it is about objects
-from a single namespace or (c) `cluster` if it spawns objects from multiple
-namespaces.
+<a name="footnote4">\[4\]</a> A scope of a request can be either
+- `resource` - if the request is about a single object
+- `namespace` - if it is about objects from a single namespace
+- `cluster` - if it spawns objects from multiple namespaces
+
+<a name="footnote5">\[5\]</a> Historically, the threshold for LISTs in
+`scope=namespace` was set to 5 seconds. It was assuming limitted size on
+objects within a single namespace. Given k8s evolution, that is not the case.
```

Comment:
Let's change to something like:

Historically, the threshold for LISTs with `scope=namespace` was set to 5 seconds. However, the threshold was chosen when Kubernetes doesn't support the scale it supports today and when individual namespace didn't contain tens of thousands (if not more) of objects of a given type. We adjusted the limits to accommodate the 
usage patterns change, given that users are fine with listing tens of thousands of objects taking more than 5 seconds.